src/main/java/org/zalando/nakadi/service/EventStreamWriterBinary.java

```diff
@@ -48,7 +68,7 @@ public int writeBatch(final OutputStream os, final Cursor cursor, final List<Str
         if (!events.isEmpty()) {
             os.write(B_EVENTS_ARRAY_BEGIN);
             for (int i = 0; i < events.size(); i++) {
-                final byte[] event = events.get(i).getBytes(StandardCharsets.UTF_8);
+                final byte[] event = extractor.apply(events.get(i)).getBytes(StandardCharsets.UTF_8);
```

Comment:
That's a possibility I didn't consider. I could duplicate this entire method in order to avoid this "extrator" thing. The advantage would be to have less type casting. On the other hand we would duplicate some code, in both binary and string classes.